force-app/main/default/classes/DMLOperations.cls

```diff
@@ -180,15 +266,20 @@ public with sharing class DMLOperations {
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
-		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
+		List<Contact> newContactsToUpsert = new List<Contact>();
+        // Loop through each Contact
+        for (Contact cont : contactsToUpsert){
+        String accName =	cont.LastName ;	// Extract the Account Name from Contact's LastName
 		// Get the account based on the name or create one if it doesn't exist
-
+        Account upsertedAccount = upsertAccount(accName); 
```

Comment:
I know the instructions say to do this, but can you do this without the upsert inside a loop?